lib/Parser/parse.cpp

```diff
@@ -7708,7 +7708,7 @@ void PidRefStack::TrackAssignment(charcount_t ichMin, charcount_t ichLim)
         {
             return;
         }
-        Assert(this->GetIchLim() >= ichLim);
+        Assert(ichMin <= this->GetIchMin() && this->GetIchLim() <= ichLim)
```

Comment:
Development is done with debug builds so makes sense to make it an error in debug builds.  The do while trick forces it to be a statement that requires a semicolon.